types/pg/index.d.ts

```diff
@@ -20,24 +22,21 @@ export interface ConnectionConfig {
     keepAlive?: boolean;
     stream?: stream.Duplex;
     statement_timeout?: false | number;
-    connectionTimeoutMillis?: number;
+    ssl?: boolean | ConnectionOptions;
+    query_timeout?: number;
     keepAliveInitialDelayMillis?: number;
 }
 
-export interface Defaults extends ConnectionConfig {
+export type ConnectionConfig = ClientConfig;
```

Comment:
I really like this approach, but I believe this breaks userland interfaces that *extend* `ConnectionConfig`. I don't know if this is an actual problem or if this is 'niche' enough to be safely changed, either way I don't think I'm the right person to decide that.
Are there DT guidelines for these kind of situations?

Reply:
Extending types seems to work, at least in the newest TS versions https://www.typescriptlang.org/play/index.html?ssl=1&ssc=1&pln=12&pc=2#code/C4TwDgpgBAglC8UDeAoK6oEMBcUB2ArgLYBGEATgNwoC+KKAlnsBQGaYDG0AQlBAB4s8AEwDOsZGgwlco4OSYBzanRQcA9njlQAbpgWYSAGwi5eiVBiy4ATAGYALABop6GVADko4aw+0gA
What are the problems you aware of?

Reply:
I guess I was wrong then, perhaps it was changed in a recent version.